### ISSUE TITLE
JSON Field wraps keys around an index key on Save when a key starts with a `.` (period).

### ISSUE THREAD

**Issue by koriolis** [payloadcms/payload#5378]
### Describe the Bug

On a JSON field any keys that contain period characters force a reformat and change the content.
When it contains a `.` (period) at the start when you press Save that key will be wrapped with a key which name is `0` and the period is removed from the original key.
If the key contains a `.` and it looks like an object path the key will be _unflattened_ like a path would be if it was a path.

Example input:
```json
{
  ".test": "abcd",
  "hello.world": 1234
}
```

Pressing Save will change the content to:
```json
{
  "0": {
    "test": "abcd"
  },
  "hello": {
    "world": 1234
  }
}
```

Adding the above payload through the REST API won't trigger a code reformat so, FWIW, my guess is the issue only happens on the Admin UI.

### To Reproduce

1. In any JSON type field in a collection, add the following content:
```json
{
  ".test": "abcd",
  "hello.world": 1234
}
```

2. Press Save.

Expected Result should be the JSON written on step 1.

Actual result is:
```json
{
  "0": {
    "test": "abcd"
  },
  "hello": {
    "world": 1234
  }
}
```

### Payload Version

2.11.2

### Adapters and Plugins
```
db-mongodb           1.4.3
bundler-webpack      1.0.6
plugin-cloud-storage 1.1.2
richtext-lexical     0.7.0
richtext-slate       1.4.0
```

**Comment by DanRibbens** [comment]
I made an attempt on this, but we have complex interactions in our Forms and I couldn't wrangle this one in the time I had yesterday. I'm posting my findings as a starting point for the next attempt to fix this. I would put this effort into v3 currently the alpha branch at this point.

The function for reducing Fields to values that get sent to the server is here: `packages/payload/src/admin/components/forms/Form/reduceFieldsToValues.ts`

It calls a utility function called unflatten that we've imported from the `flatley` package. It is murdering your dot notation by making nested objects. I attempted to fix this by bypassing the unflatten conditionally depending on the field type. This doesn't work because for some reason the `fields` type has a conditional `fieldSchema?: Field` property. I don't know why fieldSchema isn't always available, but this function is called from a great many places and at this point I'm not sure if this is the best fix. It might be better to use the form overrides, but I'm not familiar enough with that to take this further.

Here is my logic that doesn't work without fieldSchema sadly:

```
import { unflatten as flatleyUnflatten } from 'flatley'

import type { Data, Fields } from './types'

/**
 * Reduce flattened form fields (Fields) to just map to the respective values instead of the full FormField object
 *
 * @param unflatten This also unflattens the data if `unflatten` is true. The unflattened data should match the original data structure
 * @param ignoreDisableFormData - if true, will include fields that have `disableFormData` set to true, for example, blocks or arrays fields.
 *
 */
const reduceFieldsToValues = (
  fields: Fields,
  unflatten?: boolean,
  ignoreDisableFormData?: boolean,
): Data => {
  const data = {}
  const unflattenedData = {}

  Object.keys(fields).forEach((key) => {
    if (ignoreDisableFormData === true || !fields[key].disableFormData) {
      if (unflatten && fields[key].fieldSchema?.type === 'json') {
        unflattenedData[key] = fields[key].value
      } else {
        data[key] = fields[key].value
      }
    }
  })

  if (unflatten) {
    return {...flatleyUnflatten(data, { safe: true }), ...unflattenedData}
  }

  return data
}
```

**Comment by PatrikKozak** [comment]
Hey @koriolis - thank you for your patience here! We have a fix for this and will be merged shortly!

**Comment by koriolis** [comment]
Thanks a million. This will be a welcome fix!

**Comment by JarrodMFlesch** [comment]
@koriolis released in [v2.22.0](https://github.com/payloadcms/payload/releases/tag/v2.22.0)
